infoschema/builder.go

```diff
@@ -111,9 +111,16 @@ func (b *Builder) ApplyDiff(m *meta.Meta, diff *model.SchemaDiff) ([]int64, erro
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
+		if diff.Type == model.ActionTruncateTablePartition {
```

Comment:
the code is not elegant to put here, how about move it to the `applyCreateTable` and merge it with `appendAffectedIDs`. They are all about modifying the ids.

Reply:
ok, done.